src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -45,28 +46,44 @@ class DLL_EXPORT CheckinMessage
      *
-     * @param key       Key with which to encrypt the check-in payload
-     * @param counter   Check-in counter
-     * @param appData   Application Data to incorporate within the Check-in message. Allowed to be empty.
-     * @param output    Buffer in Which to store the generated payload. SUFFICIENT SPACE MUST BE ALLOCATED by the caller
-     *                  Required Buffer Size is : GetCheckinPayloadSize(appData.size())
-     * @return CHIP_ERROR
+     * @note Function requires two key handles to generate the Check-In message.
+     *       Due to PSA requirements, the same key handle cannot be used for AES-CCM and HMAC-SHA-256 operations.
+     *
+     * @param[in]  aes128KeyHandle   Key handle with which to encrypt the check-in payload with the AEAD
+     * @param[in]  hmac128KeyHandle  Key handle with which to generate the nonce the check-in payload with the HMAC
+     * @param[in]  counter           Check-in counter
+     * @param[in]  appData           Application Data to incorporate within the Check-in message. Allowed to be empty.
+     * @param[out] output            Buffer in Which to store the generated payload. SUFFICIENT SPACE MUST BE ALLOCATED by the
+     *                               caller Required Buffer Size is : GetCheckinPayloadSize(appData.size())
+     *
+     * @return CHIP_ERROR_BUFFER_TOO_SMALL if output buffer is too small
+     *         CHIP_ERROR_INVALID_ARGUMENTS if the provide arguments cannot be used to generate the Check-In message
      */
-    static CHIP_ERROR GenerateCheckinMessagePayload(Crypto::Aes128KeyHandle & key, CounterType counter, const ByteSpan & appData,
-                                                    MutableByteSpan & output);
+    static CHIP_ERROR GenerateCheckinMessagePayload(const Crypto::Aes128KeyHandle & aes128KeyHandle,
+                                                    const Crypto::Hmac128KeyHandle & hmacKeyHandle, const CounterType & counter,
+                                                    const ByteSpan & appData, MutableByteSpan & output);
 
     /**
      * @brief Parse Check-in Message payload
      *
-     * @param key       Key with which to decrypt the check-in payload
-     * @param payload   The received payload to decrypt and parse
-     * @param counter   The counter value retrieved from the payload
-     * @param appData   The optional application data decrypted. The size of appData must be at least the size of
-     *                  GetAppDataSize(payload) + sizeof(CounterType)
-     * @return CHIP_ERROR
+     * @note Function requires two key handles to generate the Check-In message.
+     *       Due to PSA requirements, the same key handle cannot be used for AES-CCM and HMAC-SHA-256 operations.
+     *
+     * @param[in]       aes128KeyHandle   Key handle with which to encrypt the check-in payload with the AEAD
+     * @param[in]       hmac128KeyHandle  Key handle with which to generate the nonce for the check-in payload with the HMAC
```

Comment:
See comments above, including both phrasing an the question about whether hmac128KeyHandle is needed here.

And we are not _generating_ the nonce here.  We are maybe verifying it?

Reply:
Update the comments to fix all copy paste erros.